modal/_utils/grpc_utils.py

```diff
@@ -198,6 +198,7 @@ async def retry_transient_errors(
     total_timeout: Optional[float] = None,  # timeout for the entire function call
     attempt_timeout_floor=2.0,  # always have at least this much timeout (only for total_timeout)
     retry_warning_message: Optional[RetryWarningMessage] = None,
+    input_plane_region: Optional[str] = None,
```

Comment:
Could we make this more general, like `metadata: list[tuple[str, str]] = []` or something?

This `retry_transient_errors` function is used in a lot of different places, so it would be nice if the arguments aren't specific to the input plane.